easybuild/tools/toolchain/mpi.py

```diff
@@ -191,7 +191,8 @@ def mpi_cmd_for(self, cmd, nr_ranks):
             tmpdir = tempfile.mkdtemp(prefix='eb-mpi_cmd_for-')
 
             # set temporary dir for mdp
-            env.setvar('I_MPI_MPD_TMPDIR', tmpdir)
+            # note: this needs to be kept *short*, to avoid mpirun failing with "socket.error: AF_UNIX path too long"
+            env.setvar('I_MPI_MPD_TMPDIR', tempfile.gettempdir())
```

Comment:
well, actually this could be too long as well, depending on what `$TMPDIR` or `--tmpdir` was set to...

We want to set this to something that avoids clashing with other users on the same system (so the default, or previously hard-set, `/tmp` isn't a good option), but it should also be kept as short as possible...